homeassistant/config_entries.py

```diff
@@ -1386,7 +1386,7 @@ async def async_step_discovery(
 
     @callback
     def async_abort(
-        self, *, reason: str, description_placeholders: dict | None = None
+        self, *, reason: str, description_placeholders: dict[str, Any] | None = None
```

Comment:
Isn't this `dict[str, str]`?

Reply:
That was initial feeling, but it is typed as `Any` in `FlowResult`
https://github.com/epenet/home-assistant-core/blob/f9aa0a7cd887456857ae7ed9466f8ded9a7cbbf5/homeassistant/data_entry_flow.py#L78

Should we amend `FlowResult` to force strings?

Reply:
I made a quick attempt to set `dict[str, str]`, and it seems a lot of components pass `Optional[str]` values, and a few components even pass `int` values.

I'm happy to start adjusting those if you feel that FlowResult should be amended.

Reply:
Looking at the codebase a bit, because of your response, it should be typed as `dict[str, str | None]` IMHO.

We should always aim for avoiding `Any` as much as possible (it kinda defeats the purpose of typing 😉 )

Reply:
Maybe `Mapping[str, str | None]` is even better?

Reply:
I don't believe this is used as other types (e.g., MappingProxy or similar), thus that seems unneeded?

Reply:
The advantage of `Mapping` is that it is covariant in the value type, which means you can pass both `dict[str, str]` and `dict[str, str | None]`

It avoids mypy errors such as `Argument "description_placeholders" to "async_show_form" of "FlowHandler" has incompatible type "Dict[str, str]"; expected "Optional[Dict[str, Optional[str]]]"`

Reply:
It would also make it easier to reject `None` in a follow-up PR

Reply:
Right using the covariant in this case makes sense 👍